docs/reference/encryption-and-security.md

```diff
@@ -0,0 +1,154 @@
+# Encryption and Security
+
+## Overview
+
+Frigg provides built-in support for data encryption to help you secure sensitive information in your integrations. The framework automatically configures AWS KMS (Key Management Service) for field-level encryption when enabled in your application definition.
+
+## Automatic KMS Configuration
+
+### Enable KMS in Your App Definition
+
+To enable automatic KMS configuration, add the `encryption` property to your App Definition:
+
+```javascript
+const appDefinition = {
+    name: 'my-frigg-app',
+    integrations: [
+        // your integrations...
+    ],
+    encryption: {
+        useDefaultKMSForFieldLevelEncryption: true
+    }
+}
+
+module.exports = appDefinition;
+```
+
+### What Happens Automatically
+
+When `useDefaultKMSForFieldLevelEncryption` is set to `true`, Frigg automatically:
+
+1. **Grants KMS Permissions**: Adds `kms:GenerateDataKey` and `kms:Decrypt` permissions to all Lambda function IAM roles
+2. **Sets Environment Variable**: Configures `KMS_KEY_ARN` environment variable for runtime access
+3. **Includes KMS Plugin**: Adds the `serverless-kms-grants` plugin to your serverless configuration
+4. **Configures Default Keys**: Uses AWS default KMS keys (`kmsKeyId: '*'`) for encryption operations
+
+### Generated Infrastructure
+
+The framework generates the following serverless configuration:
+
+```yaml
+# IAM Permissions
+provider:
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - kms:GenerateDataKey
+        - kms:Decrypt
+      Resource: 
+        - '${self:custom.kmsGrants.kmsKeyId}'
+
+# Environment Variables
+provider:
+  environment:
+    KMS_KEY_ARN: '${self:custom.kmsGrants.kmsKeyId}'
+
+# Plugins
+plugins:
+  - serverless-kms-grants
+
+# Custom Configuration
+custom:
+  kmsGrants:
+    kmsKeyId: '*'
+```
+
+## Using KMS in Your Code
+
+### Accessing the KMS Key ARN
+
+The KMS key ARN is available in your Lambda functions via environment variables:
+
+```javascript
+const kmsKeyArn = process.env.KMS_KEY_ARN;
+
+// Use with AWS SDK for encryption operations
+const { KMSClient, GenerateDataKeyCommand, DecryptCommand } = require('@aws-sdk/client-kms');
+
+const kmsClient = new KMSClient({ region: 'us-east-1' });
+```
+
+### Integration with Frigg Encrypt Module
+
+If you're using the `@friggframework/encrypt` module, it will automatically use the configured KMS key:
+
+```javascript
+const { encrypt, decrypt } = require('@friggframework/encrypt');
```

Comment:
This isn't the normal use, but fair. We also use it by default in the Credential model https://github.com/friggframework/frigg/blob/67b1ea454e7d25e4d58ff2ba08b025ca4141aa26/packages/core/module-plugin/auther.js#L172

for field-level encryption of the sensitive api keys/tokens/etc.